pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -95,6 +98,12 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 		}
 	}
 
+	for _, network := range networks {
+		if network.CIDR.Contains(net.ParseIP(config.APIVIP)) {
+			templateData.APIIntIPv6 = network.CIDR.IP.To4() == nil
+		}
+	}
```

Comment:
Is there some reason why we have to check whether the subnet is IPv4 instead of just the VIP?
```suggestion
	templateData.APIIntIPv6 = net.ParseIP(config.APIVIP).To4() == nil
```